asynch/proto/connection.py

```diff
@@ -244,15 +244,32 @@ async def receive_hello(self):
             raise UnexpectedPacketFromServerError(message)
 
     async def ping(self):
-        await self.writer.write_varint(ClientPacket.PING)
-        await self.writer.flush()
-        packet_type = await self.reader.read_varint()
-        while packet_type == ServerPacket.PROGRESS:
-            await self.receive_progress()
+        try:
+            await self.writer.write_varint(ClientPacket.PING)
+            await self.writer.flush()
             packet_type = await self.reader.read_varint()
-        if packet_type != ServerPacket.PONG:
-            msg = self.unexpected_packet_message("Pong", packet_type)
-            raise UnexpectedPacketFromServerError(msg)
+            while packet_type == ServerPacket.PROGRESS:
+                await self.receive_progress()
+                packet_type = await self.reader.read_varint()
+            if packet_type != ServerPacket.PONG:
+                msg = self.unexpected_packet_message("Pong", packet_type)
+                raise UnexpectedPacketFromServerError(msg)
+        except IndexError as e:
+            logger.debug(
+                "Ping package smaller than expected or empty. "
+                "There may be connection or network problems - "
+                "we believe that the connection is incorrect.",
+                exc_info=e,
+            )
+            return False
+        except (ConnectionError, OSError, RuntimeError) as e:
+            # If raised RuntimeError with "TCPTransport the handler is closed" - just returning false,
+            # because this is a connection loss case
+            if isinstance(e, RuntimeError) and "TCPTransport closed=True" not in str(e):
+                raise
```

Comment:
What about `raise e`?

Reply:
Yes, no problem - thanks 😃.

Of course this will work without `e` ([doc](https://docs.python.org/3.9/reference/simple_stmts.html#the-raise-statement), [stackoverflow](https://stackoverflow.com/a/45986945)), but it seems that using `e` is preferred, due to the explicit.

But it also has interesting ["features"](https://stackoverflow.com/a/36153948).

Reply:
I pushed change